pkg/util/podstore/podstore_test.go

```diff
@@ -0,0 +1,549 @@
+// Copyright 2023 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package podstore
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/rand"
+	coreinformers "k8s.io/client-go/informers/core/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/fake"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	clock "k8s.io/utils/clock/testing"
+)
+
+var (
+	refTime = time.Now()
+	pod1    = &v1.Pod{
+		Status: v1.PodStatus{
+			PodIPs: []v1.PodIP{
+				{
+					IP: "1.2.3.4",
+				},
+			},
+			Phase: v1.PodPending,
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:              "pod1",
+			Namespace:         "pod1_ns",
+			UID:               "pod1",
+			CreationTimestamp: metav1.Time{Time: refTime.Add(-5 * time.Minute)},
+		},
+	}
+	pod2 = &v1.Pod{
+		Status: v1.PodStatus{
+			PodIPs: []v1.PodIP{
+				{
+					IP: "5.6.7.8",
+				},
+			},
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "pod2",
+			Namespace: "pod2_ns",
+			UID:       "pod2",
+		},
+	}
+	pod3 = &v1.Pod{
+		Status: v1.PodStatus{
+			PodIPs: []v1.PodIP{
+				{
+					IP: "4.3.2.1",
+				},
+			},
+			Phase: v1.PodRunning,
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:              "pod3",
+			Namespace:         "pod3_ns",
+			UID:               "pod3",
+			CreationTimestamp: metav1.Time{Time: refTime.Add(-5 * time.Minute)},
+		},
+	}
+	pod4 = &v1.Pod{
+		Status: v1.PodStatus{
+			PodIPs: []v1.PodIP{
+				{
+					IP: "1.2.3.4",
+				},
+			},
+			Phase: v1.PodSucceeded,
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:              "pod1",
+			Namespace:         "pod1_ns",
+			UID:               "pod4",
+			CreationTimestamp: metav1.Time{Time: refTime.Add(-5 * time.Minute)},
+			DeletionTimestamp: &metav1.Time{Time: refTime},
+		},
+	}
+	timestampMap = map[types.UID]*podTimestamps{
+		"pod1": {CreationTimestamp: refTime},
+		"pod2": {CreationTimestamp: refTime.Add(-5 * time.Minute), DeletionTimestamp: &refTime},
+		"pod4": {CreationTimestamp: refTime.Add(-5 * time.Minute), DeletionTimestamp: &refTime},
```

Comment:
you should probably use `pod1. CreationTimestamp`, etc. for map initialization, rather than duplicate the values

Reply:
Fixed. Thanks.